python/tests/test_approximate_nearest_neighbors.py

```diff
@@ -329,7 +329,13 @@ def assert_row_equal(r1: Row, r2: Row) -> None:
             assert r1[f"query_{id_col}"] == r2[f"query_{id_col}"]
             r1_distances = r1["distances"]
             r2_distances = r2["distances"]
-            assert array_equal(r1_distances, r2_distances, tolerance)
+            if algorithm == "ivfpq":
+                # returned distances can be slightly different when running ivfpq multiple times due to quantization and randomness
+                assert array_equal(
+                    r1_distances, r2_distances, unit_tol=tolerance, total_tol=tolerance
```

Comment:
total tolerance > 0 means that some entries can be way off.   Is this needed vs just increasing unit_tol?  What would cause some entries to have huge error?

Reply:
Removed the total tolerance, the test sometimes fails, at the same place. 
Set the total tolerance to e-3 and repeat the test five times, all get passed. 
Set the total tolerance to e-4, the test fails at the same place. 

The total tolerance is needed. Returned distances are different even when running the SG class twice. Codebook difference maybe be the cause.